services/skus/credentials.go

```diff
@@ -247,12 +247,28 @@ func (s *Service) CreateOrderItemCredentials(ctx context.Context, orderID uuid.U
 		return errors.New("order item does not exist for order")
 	}
 
-	if orderItem.CredentialType == "single-use" {
+	if orderItem.CredentialType == singleUse {
 		if len(blindedCreds) > orderItem.Quantity {
 			return errors.New("submitted more blinded creds than quantity of order item")
 		}
 	}
 
+	if orderItem.CredentialType == timeLimitedV2 {
+		// check order "numPerInterval" from metadata and multiply it by the buffer added to offset
+		numPerInterval, ok := order.Metadata["numPerInterval"].(int)
+		if !ok {
+			return errors.New("bad order: numPerInterval not set in order metadata")
+		}
+		numIntervals, ok := order.Metadata["numIntervals"].(int)
```

Comment:
Same as above, we need to type-cast to `float64` here instead of `int` due to [this](https://github.com/brave-intl/bat-go/blob/master/libs/datastore/models.go#L29-L36).

```suggestion
		numIntervals, ok := order.Metadata["numIntervals"].(float64)
```

Reply:
Additionally, since the currently written type-casting (to `int`) would always fail, this place would always return an error. This could, had it been merged and triggered, have caused some issues.